src/overlays/actors/ovl_Bg_Jya_Bigmirror/z_bg_jya_bigmirror.c

```diff
@@ -26,21 +28,235 @@ const ActorInit Bg_Jya_Bigmirror_InitVars = {
+    }
+}
+
+void BgJyaBigmirror_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgJyaBigmirror_SetRoomFlag(thisx, globalCtx);
+    BgJyaBigmirror_HandleCobra(thisx, globalCtx);
+    BgJyaBigmirror_SetBombiwaFlag(thisx, globalCtx);
+    BgJyaBigmirror_HandleMirRay(thisx, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Bigmirror/func_8089394C.s")
+void BgJyaBigmirror_DrawLightBeam(Actor* thisx, GlobalContext* globalCtx) {
+    static Vec3s D_80893F4C = { 0, 0, 0 };
+    BgJyaBigmirror* this = THIS;
+    Actor* lift;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Bigmirror/func_8089399C.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_bg_jya_bigmirror.c", 435);
+    func_80093D84(globalCtx->state.gfxCtx);
+    lift = Actor_Find(&globalCtx->actorCtx, ACTOR_BG_JYA_LIFT, 1);
+    if (lift != NULL) {
+        this->liftHeight = lift->posRot.pos.y;
+    }
+    func_800D1694(this->actor.posRot.pos.x, this->actor.posRot.pos.y + 40.0f, this->actor.posRot.pos.z,
+                  &this->actor.shape.rot);
+    // Second float seems to be either this or 1613/1280 + 0.13: both numerators relate to the lift height
+    Matrix_Scale(0.1f, (this->liftHeight * -(1.0f / 1280.0f)) + (973.0f / 1280.0f + 0.63f) /* 1.3901563f */, 0.1f, MTXMODE_APPLY);
+    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_jya_bigmirror.c", 457),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_XLU_DISP++, D_0600BC70);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Bigmirror/BgJyaBigmirror_Init.s")
+    if (lift != NULL) {
+        if (1) {}
+        func_800D1694(lift->posRot.pos.x, lift->posRot.pos.y, lift->posRot.pos.z, &D_80893F4C);
+        Matrix_Scale(0.1f, 0.1f, 0.1f, 1);
+        gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_jya_bigmirror.c", 467),
+                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_XLU_DISP++, D_0600BD80);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Bigmirror/BgJyaBigmirror_Destroy.s")
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_bg_jya_bigmirror.c", 476);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Bigmirror/BgJyaBigmirror_Update.s")
+void BgJyaBigmirror_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    BgJyaBigmirror* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Bigmirror/func_80893C68.s")
+    // In first top room
+    if (this->puzzleState & 0x10) {
+        Gfx_DrawDListOpa(globalCtx, D_0600E1B0);
+        Gfx_DrawDListXlu(globalCtx, D_0600E2D0);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Bigmirror/BgJyaBigmirror_Draw.s")
+    // In statue room or two top rooms AND second cobra is solved AND first cobra is solved
+    if ((this->puzzleState & 0x38) && (this->puzzleState & 2) && (this->puzzleState & 1)) {
```

Comment:
I suggest including flags in the name to make it clear it's a field of flags. Maybe just `puzzleFlags`